includes/razorpay-subscription-webhook.php

```diff
@@ -199,8 +202,43 @@ protected function processSubscriptionSuccess($orderId, $subscription, $paymentI
      *
      * @param $orderId
      */
-    protected function processSubscriptionFailed($orderId)
+    protected function processSubscriptionFailed($orderId, $subscription, $paymentId)
     {
-        WC_Subscriptions_Manager::process_subscription_payment_failure_on_order($orderId);
+        $wcSubscription = wcs_get_subscriptions_for_order($orderId);
+        $wcSubscription = array_values($wcSubscription)[0];
+
+        $is_first_payment = ( $wcSubscription->get_completed_payment_count() < 1 ) ? true : false;
+
+        if (!$is_first_payment)
+        {
+            if ( $wcSubscription->has_status( 'active' ) )
+            {
+                $wcSubscription->update_status( 'on-hold' );
+            }
+
+            $renewal_order = $this->get_renewal_order_by_transaction_id( $wcSubscription, $paymentId );
+
+            if ( is_null( $renewal_order ) ) {
+                $renewal_order = wcs_create_renewal_order( $wcSubscription );
```

Comment:
@ccrims0n  A bit skeptic about creating a new renewal order for on payment failure.

In case of automatic renewal of subscription , if the payment fails, then it could be paid manually.

Reply:
say I attempted two failed subscription, two new renewal orders would be created. which  would be confustng,

Reply:
@ccrims0n  Cancellation logic if any change would be fixed later in a seperate pR